libseccomp/src/lib.rs

```diff
@@ -651,6 +651,92 @@ impl std::str::FromStr for ScmpArch {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
```

Comment:
Is this for being able to write `extern { fn seccomp_syscall_resolve_name_arch() -> ScmpSyscall } ` in libseccomp-sys for the future? I'm wondering whether it is necessary currently. I'm sorry if I'm wrong.

Reply:
Yes, the original idea was to be able to pass it (somehow) directly to libseccomp. However since we need a trait to continue support for passing `i32` anyway it is better to have a safe function (like `as_syscall_nr`) than an `unsafe trait` which allows use to `mem::transmute`.

Reply:
I understand. Thank you!